tsuba: Update manifest.FileNames to only return files that exist

ddn0 opened the pull request:
Entity arrays do not exist for all RDGs; for backward compatibility,
only return entity array names when they are actually present.

insertinterestingnamehere commented:
Related: https://github.com/IntelligentSoftwareSystems/Galois/pull/357. At some point it'd be nice to sync the repos again to minimize overlap of patches like this.

ddn0 commented:
Thanks for pointing on the overlap!

I've manually incorporated your changes here for now.

insertinterestingnamehere commented:
Sounds good. Thanks!

gurbinder533 reviewed (approved):
LGTM. Much cleaner.

roshandathathri commented on libgalois/include/galois/AtomicHelpers.h:
```diff
@@ -18,193 +18,74 @@
+#if __cplusplus > 201703L
+template <typename T>
+T atomicAdd(std::atomic<T>& a, const T& b) {
+  return a.fetch_add(b);
```
`a.fetch_add(b, std::memory_order_relaxed);`

roshandathathri commented on libgalois/include/galois/AtomicHelpers.h:
```diff
@@ -18,193 +18,74 @@
+template <typename T>
+T atomicAdd(std::atomic<T>& a, const T& b,
+            std::enable_if_t<std::is_integral_v<T>>* = nullptr) {
+  return a.fetch_add(b);
```
`a.fetch_add(b, std::memory_order_relaxed);`

roshandathathri commented on libgalois/include/galois/AtomicHelpers.h:
```diff
@@ -18,193 +18,74 @@
+#if __cplusplus > 201703L
+template <typename T>
+T atomicSub(std::atomic<T>& a, const T& b) {
+  return a.fetch_sub(b);
```
`a.fetch_sub(b, std::memory_order_relaxed);`

roshandathathri commented on libgalois/include/galois/AtomicHelpers.h:
```diff
@@ -18,193 +18,74 @@
+template <typename T>
+T atomicSub(std::atomic<T>& a, const T& b,
+            std::enable_if_t<std::is_integral_v<T>>* = nullptr) {
+  return a.fetch_sub(b);
```
`a.fetch_sub(b, std::memory_order_relaxed);`

roshandathathri commented on libgluon/include/galois/runtime/DataCommMode.h:
```diff
@@ -28,6 +28,8 @@
  */
 #pragma once
 
+#include <cstddef>
```
Why was this needed?

ddn0 replied:
to get the definition of `size_t`

roshandathathri commented on lonestar/analytics/distributed/sssp/sssp_pull.cpp:
```diff
@@ -27,6 +27,7 @@
 
 #include <iostream>
 #include <limits>
+#include <numeric>
```
Why not algorithm?

roshandathathri commented on libgluon/include/galois/runtime/SyncStructures.h:
```diff
@@ -446,7 +446,7 @@ class FieldFlags {
       }                                                                        \
       assert(personality == CPU);                                              \
-      galois::add(g.getData(node_id).fieldname, y);                            \
+      g.getData(node_id).fieldname = g.getData(node_id).fieldname + y;         \
```
I'm assuming you're not using `+=` because that operator might not have overloaded for that type? Forward thinking :)

ddn0 replied:
yeah; this is actually true of `std::atomic` so I had to do it.

Ian's linked change does relaxed reads when the field is an std::atomic. I'll probably add that when I follow this PR with the refactor of SyncStructures to non static functions.

roshandathathri replied:
Cool.

roshandathathri reviewed (approved):
Yay!

ddn0 committed:
tsuba: Update manifest.FileNames to only return files that exist

Entity arrays do not exist for all RDGs; for backward compatibility,
only return entity array names when they are actually present.

e-mcginnis reviewed (approved):
Looks great, thanks for fixing this!
